apis/core/v1beta1/openstackcontrolplane_webhook.go

```diff
@@ -61,9 +63,54 @@ func (r *OpenStackControlPlane) ValidateDelete() error {
 	return nil
 }
 
+func (r *OpenStackControlPlane) checkDepsEnabled(name string) bool {
+	switch name {
+	case "KeyStone":
+		return r.Spec.Mariadb.Enabled
+	case "Glance":
+		return r.Spec.Mariadb.Enabled && r.Spec.Keystone.Enabled
+	case "Cinder":
+		return r.Spec.Mariadb.Enabled && r.Spec.Rabbitmq.Enabled && r.Spec.Keystone.Enabled
+	case "Placement":
+		return r.Spec.Mariadb.Enabled && r.Spec.Keystone.Enabled
+	case "Neutron":
+		return (r.Spec.Mariadb.Enabled && r.Spec.Rabbitmq.Enabled &&
+			r.Spec.Keystone.Enabled && r.Spec.Ovn.Enabled && r.Spec.Ovs.Enabled)
```

Comment:
We probably should omit ovn and ovs now, because ovn/ovs might not be used when 3rd party SDN such as contrail is used.

Reply:
Yeah removed them. Though I was looking at the overcloud ovn default.